store/bigtable/bigtable.go

```diff
@@ -0,0 +1,470 @@
+package bigtable
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/bigtable"
+	"github.com/grafana/metrictank/mdata"
+	"github.com/grafana/metrictank/mdata/chunk"
+	"github.com/grafana/metrictank/stats"
+	"github.com/grafana/metrictank/tracing"
+	opentracing "github.com/opentracing/opentracing-go"
+	tags "github.com/opentracing/opentracing-go/ext"
+	"github.com/raintank/schema"
+	log "github.com/sirupsen/logrus"
+)
+
+const Month_sec uint32 = 60 * 60 * 24 * 28
+
+var (
+	errChunkTooSmall  = errors.New("impossibly small chunk in bigtable")
+	errStartBeforeEnd = errors.New("start must be before end.")
+	errCtxCanceled    = errors.New("context canceled")
+
+	// metric store.bigtable.get.exec is the duration of getting from bigtable store
+	btblGetExecDuration = stats.NewLatencyHistogram15s32("store.bigtable.get.exec")
+	// metric store.bigtable.get.wait is the duration of the get spent in the queue
+	btblGetWaitDuration = stats.NewLatencyHistogram12h32("store.bigtable.get.wait")
+	// metric store.bigtable.put.exec is the duration of putting in bigtable store
+	btblPutExecDuration = stats.NewLatencyHistogram15s32("store.bigtable.put.exec")
+	// metric store.bigtable.put.wait is the duration of a put in the wait queue
+	btblPutWaitDuration = stats.NewLatencyHistogram12h32("store.bigtable.put.wait")
+	// metroc store.bigtable.get.error is the count of reads that failed
+	btblReadError = stats.NewCounter32("store.bigtable.get.error")
+
+	// metric store.bigtable.chunks_per_row is how many chunks are retrieved per row in get queries
+	btblChunksPerRow = stats.NewMeter32("store.bigtable.chunks_per_row", false)
+	// metric store.bigtable.rows_per_response is how many rows come per get response
+	btblRowsPerResponse = stats.NewMeter32("store.bigtable.rows_per_response", false)
+
+	// metric store.bigtable.chunk_operations.save_ok is counter of successful saves
+	chunkSaveOk = stats.NewCounter32("store.bigtable.chunk_operations.save_ok")
+	// metric store.bigtable.chunk_operations.save_fail is counter of failed saves
+	chunkSaveFail = stats.NewCounter32("store.bigtable.chunk_operations.save_fail")
+	// metric store.bigtable.chunk_size.at_save is the sizes of chunks seen when saving them
+	chunkSizeAtSave = stats.NewMeter32("store.bigtable.chunk_size.at_save", true)
+	// metric store.bigtable.chunk_size.at_load is the sizes of chunks seen when loading them
+	chunkSizeAtLoad = stats.NewMeter32("store.bigtable.chunk_size.at_load", true)
+	// metric store.bigtable.byte_size.at_save is the number of chunk bytes saved in each bulkApply
+	chunkBytesAtSave = stats.NewMeter32("store.bigtable.chunk_bytes.at_save", true)
+)
+
+func formatRowKey(key schema.AMKey, ts uint32) string {
+	return fmt.Sprintf("%s_%d", key.MKey.String(), ts/Month_sec)
+}
+
+func formatFamily(ttl uint32) string {
+	return fmt.Sprintf("ret_%d", ttlToHours(ttl))
```

Comment:
why do we need the `ret_` prefix? do we plan to add other types of CF's later?

Reply:
Ok, i have looked over the code.   There are 2 things going on that I would like to address.
1) the intent behind using the format "ret_XX" for the retention was to avoid the ColumnFamimly name simply being a number.  WIthout advanced knowledge of what the number representing it would be confusing for anyone looking at it, which is a maintenance issue.

2) the column family names (and MaxAgePolicy) are being scaled down to the nearest hour, with inconsistent handling of ttls < 1hour. (the name is set to ret_1, but the policy sets the maxAge to 0).
The reason for this was to keep the names of the policies as short.  eg, for an 8day ttl.  "ret_192" saves a few bytes over "ret_691200".

I think both of the concerns the current approach uses can be better addressed by just setting the policy name to a duration string, ideally in the format used by raintank/dur.  eg, an 8day TTL should use the columnFamily name "8d".   And a 2year should use "2y".  
This keeps the family name small (smaller then the current approach), while being more obvious that the name represents a duration due to the use of standard units.